Lib/fontTools/varLib/builder.py

```diff
@@ -51,11 +50,12 @@ def optimizeVarData(self):
 				break
 		if not narrows:
 			break
-
-	self.VarRegionIndex = _reorderItem(self.VarRegionIndex, narrows)
-	for i in range(self.ItemCount):
-		items[i] = _reorderItem(items[i], narrows)
-
+	self.NumShorts = count - len(narrows)
```

Comment:
This line is correct if optimize is True. Otherwise, NumShorts should be set to `1+max(set(range(count)) - narrows)`.